actionview/lib/action_view/render_parser.rb

```diff
@@ -0,0 +1,237 @@
+# frozen_string_literal: true
+
+require "action_view/ripper_ast_parser"
+
+module ActionView
+  class RenderParser
+    def initialize(name, code, parser: RipperASTParser, from_controller: false)
+      @name = name
+      @code = code
+      @parser = parser
+      @from_controller = from_controller
+    end
+
+    def render_calls
+      render_nodes = @parser.parse_render_nodes(@code)
+      render_nodes.map do |method, nodes|
+        parse_method = case method
+                       when :layout
+                         :parse_layout
+                       else
+                         :parse_render
```

Comment:
We can likely send all of these through parse render. `parse_layout` was implemented for controllers